libpod/container_inspect.go

```diff
@@ -372,6 +372,16 @@ func (c *Container) generateInspectContainerConfig(spec *spec.Spec) *define.Insp
 	if spec.Process != nil {
 		ctrConfig.Tty = spec.Process.Terminal
 		ctrConfig.Env = append([]string{}, spec.Process.Env...)
+		envSecrets := c.config.ContainerMiscConfig.EnvSecrets
+
+		for k := range envSecrets {
+			for i, v := range ctrConfig.Env {
+				if strings.HasPrefix(v, k+"=") {
+					ctrConfig.Env[i] = k + "=*******"
+				}
+			}
+		}
```

Comment:
Given EnvSecrets is a map it would be more efficient to loop over ctrConfig.Env first and then just lookup the key in EnvSecrets